src/components/scheduling/SchedulingGoalsPanel.svelte

```diff
@@ -80,22 +80,54 @@
 
   async function onUpdateGoal(event: CustomEvent<SchedulingGoalPlanSpecification>) {
     const {
-      detail: { goal_metadata, specification_id, ...goalPlanSpec },
+      detail: { goal_metadata, ...goalPlanSpec },
     } = event;
 
     if ($plan) {
       await effects.updateSchedulingGoalPlanSpecification(
         $plan,
-        specification_id,
         {
           ...goalPlanSpec,
-          specification_id,
         },
         user,
       );
     }
   }
 
+  async function duplicateGoalInvocation(event: CustomEvent<SchedulingGoalPlanSpecification>) {
+    const {
+      detail: { goal_metadata, specification_id, ...goalPlanSpec },
+    } = event;
+
+    if ($plan) {
+      await effects.updateSchedulingGoalPlanSpecifications(
```

Comment:
Would `updateSchedulingGoalPlanSpecification` be more appropriate to use here since only one is being updated?

Reply:
Actually maybe this needs to be it's own separate `effect` function. If we call either, the error in the toast won't reflect the action taken. This isn't necessarily updating a spec, but more creating a new one. There's currently a function commented out in the effects file called `createSchedulingGoalPlanSpecification` that you might be able to repurpose.